src/sentry/newsletter/remote.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import absolute_import
+
+import logging
+from uuid import uuid4
+from hashlib import sha1
+
+from sentry.http import safe_urlopen
+from sentry.newsletter.base import Newsletter
+from sentry.utils import json
+
+logger = logging.getLogger('sentry.newsletter')
+
+
+def get_install_id():
+    from sentry import options
+    install_id = options.get('sentry:install-id')
+    if not install_id:
+        install_id = sha1(uuid4().bytes).hexdigest()
+        options.set('sentry:install-id', install_id)
+    return install_id
+
+
+class RemoteNewsletter(Newsletter):
+    enabled = True
+    endpoint = 'https://sentry.io/remote/newslettersubscription/'
+
+    def __init__(self, endpoint=None):
+        if endpoint is not None:
+            self.endpoint = endpoint
+
+    def update_subscription(self, user, campaign, **kwargs):
+        kwargs['campaign'] = campaign
+        kwargs['user_id'] = user.id
+        kwargs['email'] = user.email
+        kwargs['referral'] = 'onpremise'
```

Comment:
@ehfeng this is being set to `onpremise` only for sentry on-premise, not sure if you have a preference here of what this should be or if this is fine.